xarray/core/formatting.py

```diff
@@ -507,10 +508,12 @@ def array_repr(arr):
     else:
         name_str = ""
 
-    if _get_boolean_with_default("display_expand_data", default=True):
+    if _get_boolean_with_default("display_expand_data", default=True) or isinstance(
+        arr.data, MemoryCachedArray
```

Comment:
this will load the array into memory, so we'll have to do
```suggestion
        arr.variable._data, MemoryCachedArray
```